lstchain/irf/hdu_table.py

```diff
@@ -0,0 +1,394 @@
+            "OBJECT": event_table.meta["OBJECT"],
+            "OBS_MODE": event_table.meta["OBS_MODE"],
+            "N_TELS": event_table.meta["N_TELS"],
+            "TELLIST": event_table.meta["TELLIST"],
+            "INSTRUME": event_table.meta["INSTRUME"],
+        }
+        obs_index_tables.append(t_obs)
+
+    hdu_index_table = Table(hdu_index_tables)
+
+    hdu_index_header = DEFAULT_HEADER.copy()
+    hdu_index_header["HDUCLAS1"] = "INDEX"
+    hdu_index_header["HDUCLAS2"] = "HDU"
+    hdu_index_header["INSTRUME"] = t_obs["INSTRUME"]
+
+    hdu_index = fits.BinTableHDU(
+        hdu_index_table, header=hdu_index_header, name="HDU INDEX"
+    )
+    hdu_index_list = fits.HDUList([fits.PrimaryHDU(), hdu_index])
+
+    obs_index_table = QTable(obs_index_tables)
+
+    obs_index_header = hdu_index_header.copy()
+    obs_index_header["HDUCLAS2"] = "OBS"
+    obs_index_header["MJDREFI"] = event_table.meta["MJDREFI"]
+    obs_index_header["MJDREFF"] = event_table.meta["MJDREFF"]
+
+    obs_index = fits.BinTableHDU(
+        obs_index_table, header=obs_index_header, name="OBS INDEX"
+    )
+    obs_index_list = fits.HDUList([fits.PrimaryHDU(), obs_index])
+
+    return hdu_index_list, obs_index_list
+
+
+def create_event_list(data, run_number, source_name, effective_time, elapsed_time):
+    """
+    Create the event_list BinTableHDUs from the given data
+
+    Parameters
+    ----------
+        Data: DL2 data file
+                'astropy.table.QTable'
+        Run: Run number
+                Int
+        Source_name: Name of the source
+                Str
+        Effective_time: Effective time of triggered events of the run
+                Float
+        Elapsed_time: Total elapsed time of triggered events of the run
+                Float
+    Returns
+    -------
+        Events HDU:  `astropy.io.fits.BinTableHDU`
+        GTI HDU:  `astropy.io.fits.BinTableHDU`
+        Pointing HDU:  `astropy.io.fits.BinTableHDU`
+    """
+
+    tel_list = np.unique(data["tel_id"])
+
+    # Timing parameters
+    t_start = data["dragon_time"].value[0]
+    t_stop = data["dragon_time"].value[-1]
+    time = Time(data["dragon_time"], format="unix", scale="utc")
+    date_obs = time[0].to_value("iso", "date")
+
+    # Position parameters
+    reco_alt = data["reco_alt"]
+    reco_az = data["reco_az"]
+    pointing_alt = data["pointing_alt"]
+    pointing_az = data["pointing_az"]
+
+    src_sky_pos = SkyCoord(
+        alt=reco_alt, az=reco_az, frame=AltAz(obstime=time, location=location)
+    ).transform_to(frame="icrs")
+    tel_pnt_sky_pos = SkyCoord(
+        alt=pointing_alt[0],
+        az=pointing_az[0],
+        frame=AltAz(obstime=time[0], location=location),
+    ).transform_to(frame="icrs")
+
+    try:
+        object_radec = SkyCoord.from_name(source_name)
+    except Exception:
+        log.error("Name resolve error in finding the Object in Sesame")
+        object_radec = SkyCoord(tel_pnt_sky_pos.icrs)
```

Comment:
I don't think you should just fallback here to the pointing (e.g. ON observations).

Like I said before, I don't think it is  a good idea at all to rely on this resolving feature. Especially since I learned that the compute nodes in the La Palma do not have direct internet access, it would be better to just add traitlets for `source_ra` and `source_dec` (as strings, to be passed to `SkyCoord(ra=self.source_ra, dec=self.source_dec, frame='icrs')`

Reply:
I suppose I can just replace the `source_name` traitlet with the Ra/Dec values